oem/firstvoices/android/build.sh

```diff
@@ -1,97 +1,119 @@
-  echo "  -no-update              Don't copy or build the Keyman Engine library in (assumes already present)"
-  echo "  -lib-build              Force rebuild of the Keyman Engine library"
-  echo "  -no-lib-build           Only rebuild the Keyman Engine library if it doesn't exist in /android"
-  echo "  -download-resources     Download fv_all.kmp from downloads.keyman.com"
-  echo "                          (dictionaries will be downloaded within the app)"
-  echo ""
-  exit 1
-}
+# This script runs from its own folder
+cd "$THIS_SCRIPT_PATH"
+
+# ################################ Main script ################################
+
+# Definition of global compile constants
+
+CONFIG="release"
+BUILD_FLAGS="build -x lint -x test"                     # Gradle build w/o test
+TEST_FLAGS="-x assembleRelease lintRelease testRelease" # Gradle test w/o build
+DAEMON_FLAG=
+
+builder_describe "Builds FirstVoices for Android app." \
+  "@../../../android/KMEA" \
+  "clean" \
+  "configure" \
+  "build" \
+  "test             Runs lint and tests." \
+  "--ci             Don't start the GRadle daemon. For CI" \
+  "--upload-sentry  Upload to sentry"
+
+# parse before describe_outputs to check debug flags
+builder_parse "$@"
+
+if builder_has_option --debug; then
+  builder_heading "### Debug config ####"
+  CONFIG="debug"
+  BUILD_FLAGS="assembleDebug -x lint -x test"
+  TEST_FLAGS="-x assembleDebug lintDebug testDebug"
+fi  
 
-export TARGET=FirstVoices
-KEYBOARD_PACKAGE_ID="fv_all"
-KEYBOARDS_TARGET="$KEYMAN_ROOT/oem/firstvoices/android/app/src/main/assets/${KEYBOARD_PACKAGE_ID}.kmp"
-KEYBOARDS_CSV="$KEYMAN_ROOT/oem/firstvoices/keyboards.csv"
-KEYBOARDS_CSV_TARGET="$KEYMAN_ROOT/oem/firstvoices/android/app/src/main/assets/keyboards.csv"
-
-# This build script assumes that the https://github.com/keymanapp/keyboards repo is in
-# the same parent folder as this repo, with the default name 'keyboards'
-
-PARAM_DEBUG=
-PARAM_NO_DAEMON=
-PARAM_NO_UPDATE=
-PARAM_LIB_BUILD=
-PARAM_NO_LIB_BUILD=
-DO_KEYBOARDS_DOWNLOAD=false
-
-while [[ $# -gt 0 ]] ; do
-  key="$1"
-  case $key in
-    -download-resources)
-      DO_KEYBOARDS_DOWNLOAD=true
-      ;;
-    -h|-\?)
-      display_usage
-      ;;
-    -debug)
-      PARAM_DEBUG=-debug
-      ;;
-    -no-daemon)
-      PARAM_NO_DAEMON=-no-daemon
-      ;;
-    -no-update)
-      PARAM_NO_UDPATE=-no-update
-      ;;
-    -lib-build)
-      PARAM_LIB_BUILD=-lib-build
-      ;;
-    -no-lib-build|-lib-nobuild)
-      PARAM_NO_LIB_BUILD=-no-lib-build
-      ;;
-  esac
-  shift
-done
-
-# Verify default keyboard package exists
-if [[ ! -f "$KEYBOARDS_TARGET" || ! -f "$KEYBOARDS_CSV_TARGET" ]]; then
-  echo "$KEYBOARDS_TARGET and $KEYBOARDS_CSV_TARGET required. Will download the latest version"
-  DO_KEYBOARDS_DOWNLOAD=true
+ARTIFACT="app-$CONFIG.apk"
+
+builder_describe_outputs \
+  configure     app/libs/keyman-engine.aar \
+  build:app     app/build/outputs/apk/$CONFIG/firstvoices${ARTIFACT}
```

Comment:
Is this right? It doesn't look right...

Reply:
ah, I'll fix it so
```
ARTIFACT="firstvoices-$VERSION.apk"
  build          app/build/outputs/apk/$CONFIG/${ARTIFACT}
  ```

Even though this was wrong, the actual output was controlled in build.gradle (compiles to firstvoices-17.0.60.apk)